planner/core/logical_plan_builder.go

```diff
@@ -6826,6 +6827,11 @@ func (b *PlanBuilder) buildWindowFunctionFrameBound(_ context.Context, spec *ast
 	if err != nil {
 		return nil, err
 	}
+	funcName = ast.Cast
+	bound.CastFuncs[0], err = expression.NewFunctionBase(b.ctx, funcName, bound.CalcFuncs[0].GetType(), col)
```

Comment:
How about when there is no need to cast?

Reply:
> How about when there is no need to cast?

okk, I put the col expression to CastFuncs when there is no need to cast.